test/distrib/cpp/run_distrib_test_cmake_module_install_pkgconfig.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# Copyright 2017 gRPC authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -ex
+
+cd "$(dirname "$0")/../../.."
+
+echo "deb http://archive.debian.org/debian jessie-backports main" | tee /etc/apt/sources.list.d/jessie-backports.list
+echo 'Acquire::Check-Valid-Until "false";' > /etc/apt/apt.conf
+sed -i '/deb http:\/\/deb.debian.org\/debian jessie-updates main/d' /etc/apt/sources.list
+apt-get update
+apt-get install -t jessie-backports -y libssl-dev pkg-config wget
+
+# Install CMake 3.16
+wget -q -O cmake-linux.sh https://github.com/Kitware/CMake/releases/download/v3.16.1/cmake-3.16.1-Linux-x86_64.sh
+sh cmake-linux.sh -- --skip-license --prefix=/usr
+rm cmake-linux.sh
+
+# Install gRPC with all dependencies
+mkdir -p "cmake/build"
+pushd "cmake/build"
+cmake ../..
```

Comment:
Is openssl or boringssl used?  If you don't specify 'GRPC_SSL_PROVIDER=package',   boringssl is going to be used (in which case you don't need to install libssl-dev above) - but I though that won't build actually.

Reply:
See the failure https://source.cloud.google.com/results/invocations/0b684d72-e21f-46f0-8a27-56140c7812da/targets/github%2Fgrpc/tests

I think you should add all the cmake defines from here: https://github.com/grpc/grpc/blob/eba60d8dbe4099c34b8097b2c89998d4484740ac/test/distrib/cpp/run_distrib_test_cmake_module_install.sh#L34

Reply:
I updated the PR as you suggested to pass the build but it seems there is a boringssl build issue. Do you know why this happened? 

```
/var/local/git/grpc/third_party/boringssl/crypto/fipsmodule/rand/urandom_test.cc:213:12: error: '__NR_getrandom' was not declared in this scope
       case __NR_getrandom:
            ^
/var/local/git/grpc/third_party/boringssl/crypto/fipsmodule/rand/urandom_test.cc:219:26: error: 'GRND_NONBLOCK' was not declared in this scope
           if (regs.rdx & GRND_NONBLOCK) {
                          ^
/var/local/git/grpc/third_party/boringssl/crypto/fipsmodule/rand/urandom_test.cc: In function 'std::vector<Event> TestFunctionPRNGModel(unsigned int)':
/var/local/git/grpc/third_party/boringssl/crypto/fipsmodule/rand/urandom_test.cc:348:37: error: 'GRND_NONBLOCK' was not declared in this scope
   ret.push_back(Event::GetRandom(1, GRND_NONBLOCK));
                                     ^
/var/local/git/grpc/third_party/boringssl/crypto/fipsmodule/rand/urandom_test.cc: In lambda function:
/var/local/git/grpc/third_party/boringssl/crypto/fipsmodule/rand/urandom_test.cc:398:41: error: 'GRND_NONBLOCK' is not captured
       ret.push_back(Event::GetRandom(1, GRND_NONBLOCK));
                                         ^
/var/local/git/grpc/third_party/boringssl/crypto/fipsmodule/rand/urandom_test.cc: In lambda function:
/var/local/git/grpc/third_party/boringssl/crypto/fipsmodule/rand/urandom_test.cc:406:55: error: 'GRND_NONBLOCK' is not captured
       ret.push_back(Event::GetRandom(len, block ? 0 : GRND_NONBLOCK));
                                                       ^
```

I prefer to make this simpler to reproduce what users will do first with gRPC and cmake, basically `cmake .. & make install`. Although this isn't optimal (debug build...), it's supposed to be working.

Reply:
Yes, I certainly understand that users would expect `cmake .. & make install` to just work, but there are good reasons why it doesn't. Also, it's an inherently flawed way to install any C++ software, because you can run into problems with dependencies (if your project tries to depend on same dependencies as grpc does, you'll have issues).
This is also why I am not thrilled about adding all the dependencies we added recently (they make installation experience worse).